tests/message/test_tasks.py

```diff
@@ -46,28 +46,80 @@ def test_extra_data(self):
+        self.assertTrue(wtct.provider_ethereum_public_key)
+        self.assertTrue(is_checksum_address(wtct.provider_ethereum_address))
 
     def test_ethereum_address_provider(self):
-        msg = self.FACTORY()
-        provider_public_key = decode_hex(msg.provider_ethereum_public_key)
+        wtct = self.FACTORY()
+        provider_public_key = decode_hex(wtct.provider_ethereum_public_key)
 
-        self.assertEqual(msg.provider_ethereum_address,
+        self.assertEqual(wtct.provider_ethereum_address,
                          to_checksum_address(
                              '0x' + sha3(provider_public_key)[12:].hex()))
 
     def test_ethereum_address(self):
-        msg = self.FACTORY()
-        serialized = shortcuts.dump(msg, None, None)
+        wtct = self.FACTORY()
+        serialized = shortcuts.dump(wtct, None, None)
         msg_l = shortcuts.load(serialized, None, None)
         self.assertEqual(len(msg_l.provider_ethereum_address), 2 + (20*2))
 
     def test_task_id(self):
         wtct = self.FACTORY()
         self.assertEqual(wtct.task_id, wtct.task_header.task_id)
 
+    def test_num_subtasks_default_1(self):
+        wtct = message.tasks.WantToComputeTask()
+        self.assertEqual(wtct.num_subtasks, 1)
+
+    def test_num_subtasks_none_1(self):
+        wtct = message.tasks.WantToComputeTask(num_subtasks=None)
+        self.assertEqual(wtct.num_subtasks, 1)
+        self.assertIsInstance(wtct.num_subtasks, int)
+
+    def test_num_subtasks_more_than_1(self):
+        wtct = message.tasks.WantToComputeTask(num_subtasks=17)
+        wtct_1 = helpers.dump_and_load(wtct)
+        self.assertEqual(wtct.num_subtasks, 17)
+        self.assertEqual(wtct.num_subtasks, wtct_1.num_subtasks)
+
+    def test_num_subtasks_non_int_raises(self):
+        wtct = message.tasks.WantToComputeTask(num_subtasks=3.14)
+        serialized = shortcuts.dump(wtct, None, None)
+        with self.assertRaisesRegex(
+            exceptions.FieldError,
+            r"Should be a integer \[num_subtasks:3.14\]"
+        ):
+            shortcuts.load(serialized, None, None)
+
+    def test_num_subtasks_zero_raises(self):
+        wtct = message.tasks.WantToComputeTask(num_subtasks=0)
+        serialized = shortcuts.dump(wtct, None, None)
+        with self.assertRaisesRegex(
+            exceptions.FieldError,
+            r"Should be a positive integer \[num_subtasks:0\]"
+        ):
+            shortcuts.load(serialized, None, None)
+
+    def test_num_subtasks_negative_raises(self):
+        wtct = message.tasks.WantToComputeTask(num_subtasks=-7)
+        serialized = shortcuts.dump(wtct, None, None)
+        with self.assertRaisesRegex(
+            exceptions.FieldError,
+            r"Should be a positive integer \[num_subtasks:-7\]"
+        ):
+            shortcuts.load(serialized, None, None)
+
+    def test_num_subtasks_int(self):
+        wtct = self.FACTORY()
+        self.assertIsInstance(wtct.num_subtasks, int)
```

Comment:
These tests are related to validator not this field. IMO only testing whether validator is called is appropriate here.

Reply:
those tests are here to check not only if validator was called... it is a mere lo level detail. Those tests are checking if validation occurs and behaves as intended.

Reply:
DRY? :) Test validator once, and then test whether validator was called?

Reply:
I've added missing tests for `validate_integer` and new tests for `validate_positive_integer`, but testing it also here with regex message to be sure correct field raises exception.

Also look at line [398](https://github.com/golemfactory/golem-messages/pull/334/files#diff-1ee15a75a9c8af4014de85c3cfa2bd78R398) below, is it DRY?